File: 0821-shortest-distance-to-a-character/0821-shortest-distance-to-a-character.rb
# @param {String} s
# @param {Character} c
# @return {Integer[]}
def shortest_to_char(s, c)
  res=[]
  nums = s.each_char.with_index.filter_map {|char, i| i if char == c}
  s.chars.each_index do |num|
    res << (num -  nums.min_by {|x| (x - num).abs}).abs
  end
  res
end

File: 0821-shortest-distance-to-a-character/0821-shortest-distance-to-a-character.py
class Solution:
    def shortestToChar(self, s: str, c: str) -> List[int]:
        res = []
        nums = [i for i, char in enumerate(s) if char == c]
        
        for i in range(len(s)):
            closest = min(nums, key=lambda x: abs(x - i))
            res.append(abs(i - closest))
        
        return res
        